tests/unit/plugins/inventory/test_flightctl.py

```
@@ -0,0 +1,1112 @@
+        mock_setup_conn.return_value = {'api_key': 'fake_key', 'url': 'fake_url'}
+        mock_get_devices.side_effect = Exception("API connection error")
+
+        # Verify that the exception is caught and handled
+        with self.assertRaises(Exception) as context:
+            self.inventory.parse(self.mock_ansible_inventory, self.mock_loader, self.config_path)
+
+        self.assertIn("API connection error", str(context.exception))
+
+    @patch('plugins.inventory.flightctl._get_devices_by_labels_and_fields')
+    @patch('plugins.inventory.flightctl._prepare_additional_groups_info')
+    def test_populate_inventory_additional_groups(self, mock_prepare_additional_groups_info,
```

Ruff (0.8.2)

Redefinition of unused `test_populate_inventory_additional_groups` from line 464

(F811)